lib/octokit/response/base_middleware.rb

```diff
@@ -0,0 +1,8 @@
+require 'faraday'
+
+module Octokit
+  module Response
+    # In Faraday 2.x, Faraday::Response::Middleware was removed
+    BaseMiddleware = defined?(Faraday::Response::Middleware) ? Faraday::Response::Middleware : Faraday::Middleware
```

Comment:
Worth noting that `Faraday::Middleware` can be used as far back as v1.2.0: https://github.com/lostisland/faraday/commit/d56742a174fe9f2f9f3d46438324fcac0d2381f7